sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -521,38 +522,99 @@ class Analyzer(
    */
   object ResolveSortReferences extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case s @ Sort(ordering, global, p @ Project(projectList, child))
-          if !s.resolved && p.resolved =>
-        val (newOrdering, missing) = resolveAndFindMissing(ordering, p, child)
+      case s @ Sort(_, _, a: Aggregate) if a.resolved =>
```

Comment:
```
select sum(product) over () from windowData where product > 3
group by product having avg(month) > 0
order by avg(area), product
```
This is not resolvable unless we reimplement the rule `ResolveAggregateFunctions`. So far, the rule assumes Sort and Aggregate are adjacent. If you read the plan, you will understand this is not true. This is not a simple change. Do you want me to do it in a separate JIRA?

```
'Sort ['avg('area) ASC,'product ASC], true
+- Project [_c0#61L]
   +- Project [_w0#62L,_c0#61L,_c0#61L]
      +- Window [_w0#62L], [(sum(_w0#62L),mode=Complete,isDistinct=false) windowspecdefinition(ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING) AS _c0#61L]
         +- Project [_w0#62L]
            +- Filter havingCondition#63: boolean
               +- Aggregate [product#2], [cast(((avg(cast(month#0 as bigint)),mode=Complete,isDistinct=false) > cast(0 as double)) as boolean) AS havingCondition#63,cast(product#2 as bigint) AS _w0#62L]
                  +- Filter (product#2 > 3)
                     +- Subquery windowdata
                        +- LogicalRDD [month#0,area#1,product#2], MapPartitionsRDD[1] at apply at Transformer.scala:22
```